controllers/vspheremachine_controller.go

```diff
@@ -242,6 +242,21 @@ func (r *machineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_
 		return reconcile.Result{}, nil
 	}
 
+	// If the VSphereCluster has been previously set as an ownerReference remove it. This may have been set in an older
+	// version of CAPV to prevent VSphereMachines from being orphaned, but is no longer needed.
+	// For more info see: https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/issues/2054
+	// TODO: This should be removed in a future release of CAPV.
+	machineContext.GetVSphereMachine().SetOwnerReferences(
+		clusterutilv1.RemoveOwnerRef(
+			machineContext.GetVSphereMachine().GetOwnerReferences(),
+			metav1.OwnerReference{
```

Comment:
Q: WDYT adding an if with a nil check for infrastructureRef around this block?

I think it usually should never happen that at this point the infraRef in the Cluster is nil, but it's CAPV.. so I guess everything is possible. Just want to avoid panics here